File: src/main/java/de/tectoast/emolga/utils/sql/managers/ItemDataManager.java
package de.tectoast.emolga.utils.sql.managers;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.utils.sql.base.DataManager;
import de.tectoast.emolga.utils.sql.base.columns.StringColumn;

public class ItemDataManager extends DataManager {

    final StringColumn NAME = new StringColumn("name", this);
    final StringColumn DESCRIPTION = new StringColumn("description", this);

    public ItemDataManager() {
        super("itemdata");
        setColumns(NAME, DESCRIPTION);
    }

    public String getData(String abiname) {
        return DESCRIPTION.retrieveValue(NAME, Command.toSDName(abiname));
    }
}

File: src/main/kotlin/de/tectoast/emolga/utils/sql/managers/ItemDataManager.kt
package de.tectoast.emolga.utils.sql.managers

import de.tectoast.emolga.commands.Command.Companion.toSDName
import de.tectoast.emolga.utils.sql.base.DataManager
import de.tectoast.emolga.utils.sql.base.columns.StringColumn

object ItemDataManager : DataManager("itemdata") {
    private val NAME = StringColumn("name", this)
    private val DESCRIPTION = StringColumn("description", this)

    init {
        setColumns(NAME, DESCRIPTION)
    }

    fun getData(name: String): String? {
        return DESCRIPTION.retrieveValue(NAME, toSDName(name))
    }
}